### PR TITLE
feat: support pkgInfo.egg.require

### PR DESCRIPTION
**Pull request by atian25**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

support:

```js
{
  "name": "example",
  "egg": {
    "require": [
       "./inject.js"
    ]
  }
}
```

though already support [pkgInfo.eggScriptConfig.require](https://github.com/eggjs/egg-scripts/blob/05ac75a53814352f0cef94b902833cc01a81f022/lib/command.js#L69) , but it need to provide full path.

and more important is should act like egg-bin: https://github.com/eggjs/egg-bin/blob/8666e9eb9ce5016ac61af9f542b5518537a90a6b/lib/command.js#L84

### CODE REVIEW

**Review by hyj1991 — APPROVED** [review]
+1

File: lib/command.js [hunk 1]
```diff
@@ -38,18 +44,28 @@ class Command extends BaseCommand {
...
+      // read `egg.typescript` from package.json
+      if (eggInfo && eggInfo.typescript) {
         argv.sourcemap = true;
       }
 
+      // read `egg.require` from package.json
+      if (eggInfo && eggInfo.require && Array.isArray(eggInfo.require)) {
+        execArgvObj.require = execArgvObj.require || [];
+        eggInfo.require.forEach(injectScript => {
+          execArgvObj.require.push(path.resolve(baseDir, injectScript));
```
Comment (atian25) [inline lib/command.js:63]: 这里不应该 resolve 的，因为一般的用法是：

```json
{
  "name": "example",
  "version": "1.0.0",
  "egg": {
    "require": [
      "babel/register.js"
    ]
  }
}
```

加了 resolve 后，就得写成 `./node_modules/babel/register.js`

Comment (atian25) [reply lib/command.js:63]: 其实 egg-bin 和 egg-scripts 这段逻辑，直接下沉到 egg-cluster 可能更适合。不过先修复这个再说

**Comment by atian25** [comment]
这个 PR 需要回滚，之前用户心智的 `pkgInfo.egg.require` 是用于开发期的，如 babel/register 这些，不能在运行期加载的。

直接用 https://github.com/eggjs/egg-scripts/blob/fe179fda909cd7eb5b6497357202185a4ecf5ec6/lib/command.js#L68 吧

**Pull request merged**